### PR TITLE
Refactor: prod 404page, og:url dynamic, list page

### LINKED ISSUE #49: create `list.html` for view all contents list
- assignee: ?
- [x] rename current `list.html` to `../index.html`
- [x] update `list.html` for showing contents list
- #48

### PR DESCRIPTION
**Pull request by vankichi**
- Refactor ♻️ 
  - make og:url dynamic
  - create `ipfs-404.html` when running `hugo --environment=production`
  - change `list.html` template: show the list of document, move previous `layouts/_default/list.html` to `layouts/_default/index.html`
- Join new user 🎉 
  - japansearch: https://jpsearch.go.jp
- close #49

### CODE REVIEW

**Comment by kpango** [comment]
could you please deploy this pr to staging?

**Comment by vankichi** [comment]
@kpango already deployed via Github Actions 😎

**Comment by kpango** [comment]
JPS-180x150-dark
could you please add this image to Vald Users?

**Comment by kpango** [comment]
and link is here (https://jpsearch.go.jp)

**Comment by taisuou** [comment]
I think list page design is good already.
I have nothing to do about this page.

File: themes/vald/layouts/_default/list.html [hunk 1]
```diff
@@ -1,243 +1,52 @@
...
-        <picture class="bg__wrap--dark">
-          <source class="bg__image" type="image/webp" srcset="{{ .Site.BaseURL }}/images/concept_bg_dark.png.webp" />
-          <img class="bg__image" src="{{ .Site.BaseURL }}/images/concept_bg_dark.png" alt="" >
-        </picture>
-      </div>
-    </div>
-  </div>
+    <h2>Overview</h2>
+      <p>
+        Overview shows the concept of Vald and mentinos the top level of Vald.
```
Comment (hlts2) [inline themes/vald/layouts/_default/list.html:14]:
```suggestion
        Overview shows the concept of Vald and mentions the top level of Vald.
```

File: themes/vald/layouts/_default/list.html [hunk 2]
```diff
@@ -1,243 +1,52 @@
...
-        <ul class="about__list about__list--center">
-          <li class="about__item about__item--single"><a href="https://yahoo.co.jp/" target="_blank"><img class="about__logo" src="{{ .Site.BaseURL }}/images/yahoojapan.svg" alt="yahoojapan"></a></li>
-        </ul>
-      </div>
-    </div>
-  </div>
+    <h2>Usecase</h2>
+      <p>
+        Usecase supports you image how to use Vald for your services.<br>
+        Also, you can get the introduction exapmles.
```
Comment (hlts2) [inline themes/vald/layouts/_default/list.html:34]:
```suggestion
        Also, you can get the introduction examples.
```

File: themes/vald/layouts/_default/list.html [hunk 3]
```diff
@@ -1,243 +1,52 @@
...
+      </p>
+      {{ partial "list.html" (dict "menu" .Site.Menus.usecase "current" $current) -}}
+
+    <h2>Contributing</h2>
+      <p>
+        We are welcome to contributing for Vald even not as developper.<br>
```
Comment (hlts2) [inline themes/vald/layouts/_default/list.html:40]:
```suggestion
        We are welcome to contribute for Vald even not as developer.<br>
```

File: themes/vald/layouts/_default/index.html [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        </li>
+        <li class="features__item">
+          <div class="features__title">
+            <picture class="card-image">
+              <source type="image/webp" srcset="{{ .Site.BaseURL }}/images/features_icon_distributed.png.webp" />
+              <img src="{{ .Site.BaseURL }}/images/features_icon_distributed.png" alt="" width="48px" height="48px">
+            </picture>
+            <h3 class="features__index distributed">Distributed Indexing</h3>
+          </div>
+          <p class="features__text">Vald distribute vector index to multiple agent, each agent stores different index.</p>
```
Comment (hlts2) [inline themes/vald/layouts/_default/index.html:116]:
```suggestion
          <p class="features__text">Vald distributes vector index to multiple agent, each agent stores different index.</p>
```

File: themes/vald/layouts/_default/list.html [hunk 5]
```diff
@@ -1,243 +1,52 @@
...
-            <h3 class="features__index multi-language">Multi language supported</h3>
-          </div>
-          <p class="features__text">Golang, Java, Nodejs and python is supported.</p>
-        </li>
-      </ul>
-    </div>
-  </div>
+    <h2>Tutorial</h2>
+      <p>
+        Tutorial takes you Vald World!!!<br>
```
Comment (kevindiu) [inline themes/vald/layouts/_default/list.html:20]:
```suggestion
        Tutorial takes you to Vald World!!!<br>
```

Comment (vankichi) [reply themes/vald/layouts/_default/list.html:20]: It may wrong meaning

**Comment by vankichi** [comment]
@kpango added jpsearch 🎉

**Comment by kpango** [comment]
LGTM

**Review by hlts2 — APPROVED** [review]
LGTM

**Pull request merged**